.github/workflows/ci.yml

```diff
@@ -1,57 +1,45 @@
-name: ExceptionPage CI
+name: CI
 
 on:
+  pull_request:
   push:
     branches: [master]
-  pull_request:
-    branches: "*"
+  schedule:
+    - cron: "0 3 * * 1" # Every monday at 3 AM
 
 jobs:
-  check_format:
+  test:
     strategy:
       fail-fast: false
       matrix:
-        crystal_version:
-          - 0.36.1
-          - 1.0.0
-        experimental:
-          - false
-    runs-on: ubuntu-latest
-    continue-on-error: ${{ matrix.experimental }}
-    container: crystallang/crystal:${{ matrix.crystal_version }}
-    steps:
-      - uses: actions/checkout@v2.3.4
-      - name: Install shards
-        run: shards install
-      - name: Format
-        run: crystal tool format --check
-      - name: Lint
-        run: ./bin/ameba
+        os: [ubuntu-latest]
+        crystal: [0.36.1, latest, nightly]
+    runs-on: ${{ matrix.os }}
 
-  specs:
-    strategy:
-      fail-fast: false
-      matrix:
-        crystal_version:
-          - 0.36.1
-          - 1.0.0
-        experimental:
-          - false
-    runs-on: ubuntu-latest
-    continue-on-error: ${{ matrix.experimental }}
-    container: crystallang/crystal:${{ matrix.crystal_version }}
     steps:
-      - uses: actions/checkout@v2.3.4
-      - name: Install Chromedriver
-        run: |
-          apt-get update
-          apt-get -yqq install chromium-chromedriver
-      - name: Cache Crystal
-        uses: actions/cache@v2.1.4
+      - name: Install Crystal
+        uses: oprypin/install-crystal@v1
         with:
-          path: ~/.cache/crystal
-          key: ${{ runner.os }}-crystal
-      - name: Install shards
+          crystal: ${{ matrix.crystal }}
+
+      - name: Install ChromeDriver
+        run: |
+          sudo apt-get update
+          sudo apt-get -yqq install chromium-chromedriver
+
+      - name: Download source
+        uses: actions/checkout@v2
+
+      - name: Install dependencies
         run: shards install
-      - name: Run tests
+        env:
+          SHARDS_OPTS: --ignore-crystal-version
+
+      - name: Run specs
         run: crystal spec
+
+      - name: Check formatting
+        run: crystal tool format --check
```

Comment:
It is not good to check formatting on multiple Crystal versions at once (especially nightly, and it can be annoying enough on its own). There can be cases where it's impossible to satisfy both versions' opinion at once

Reply:
oh, that's a good point. Luckily I haven't run in to that yet, but I can see how that might happen.

Reply:
I disagree. I prefer to know when the formatting is going to break in the next release.

Reply:
We can probably just leave as is for now. When that moment comes that it fails on one version, but not another, that can be a PR to adjust. I don't want to pre-optimize for an edge case we might not hit for a while.

Reply:
Personally, I've never hit it so far and I'm using GHA matrix runners for several months now 🤷‍♂️

Reply:
If you want to have to haphazardly fix the CI config in your projects when this happens, you're welcome to do so. I have pointed out that there is a real problem with that, though. It's easy to write the config correctly in the first place and no disadvantage to it either. Please just change the approach going forward. (Namely, check format only on latest version). Don't sign up others for this surprise.

I'm really not adding extra conditions into the example config just for the fun of it
https://oprypin.github.io/install-crystal/configurator.html

Reply:
@oprypin I don't see the "problem", as I mentioned already I consider this as a feature.

Reply:
@Sija 
> I prefer to know when the formatting is going to break in the next release.

Why do you need to know that? Ill-formatted code still compiles, so there's no need to rush changing the formatting *before* the release. Especially considering, that nightlies could contain experimental code that gets reverted back and forth if it doesn't work out. I really wouldn't recommend Crystal projects to follow nightly formatter.

Reply:
@straight-shoota I was thinking about the formatter bugs - which are plenty.